src/sentry/static/sentry/app/views/organizationIncidents/details/header.jsx

```diff
@@ -10,14 +10,58 @@ import InlineSvg from 'app/components/inlineSvg';
 import {PageHeader} from 'app/styles/organization';
 import space from 'app/styles/space';
 import SubscribeButton from 'app/components/subscribeButton';
+import DropdownControl from 'app/components/dropdownControl';
+import MenuItem from 'app/components/menuItem';
+import Access from 'app/components/acl/access';
+import DropdownButton from 'app/components/dropdownButton';
+
+import Status from '../status';
+import {isOpen} from '../utils';
 
 export default class IncidentHeader extends React.Component {
   static propTypes = {
     incident: SentryTypes.Incident,
     params: PropTypes.object.isRequired,
     onSubscriptionChange: PropTypes.func.isRequired,
+    onStatusChange: PropTypes.func.isRequired,
   };
 
+  renderStatus() {
+    const {incident, onStatusChange} = this.props;
+
+    const isIncidentOpen = isOpen(incident);
+
+    return (
+      <Access
+        access={['org:write']}
+        renderNoAccessMessage={() => <Status incident={incident} />}
+      >
+        <DropdownControl
+          button={
+            // eslint-disable-next-line no-shadow
+            ({getActorProps, isOpen}) => (
+              <DropdownButton {...getActorProps()} isOpen={isOpen}>
+                <Status incident={incident} />
+              </DropdownButton>
+            )
+          }
+          menuWidth="160px"
+        >
+          {isIncidentOpen && (
+            <StyledMenuItem onSelect={onStatusChange}>
+              {t('Close this incident')}
+            </StyledMenuItem>
+          )}
+          {!isIncidentOpen && (
+            <StyledMenuItem onSelect={onStatusChange}>
+              {t('Reopen this incident')}
+            </StyledMenuItem>
+          )}
+        </DropdownControl>
```

Comment:
Why are we giving the user a dropdown with only a single option in it? Could this be a toggle/slider instead?

Reply:
@ckj Any thoughts? I'm also not 100% sure we'll always only have one action in here - since this wasn't in the mocks or specs took a guess at these 2 options.

Reply:
I would say it adds a bit of weight to the action (in a good way). It makes the decision a bit more intentional and less change of accidentally closing an incident.